pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -113,6 +130,7 @@ public int filterEntriesForConsumer(Optional<EntryWrapper[]> entryWrapper, int e
         long totalBytes = 0;
         int totalChunkedMessages = 0;
         int totalEntries = 0;
+        FilterContext filterContext = new FilterContext();
```

Comment:
Should reuse the FilterConext here? otherwise we will introduce more objects during the message dispatching

Reply:
And if the broker don't have a filter, we should not create such objects? we can use a FilterContextDisabled to instead?